### PR TITLE
Automate NPM package release process

### PR DESCRIPTION
**Pull request by rossedfort**
## What was changed

- modify existing npm github action to create a PR with a version bump
- add new github action to publish the package and create a release
## Why?

Manual processes are not fun
## Checklist

1. Closes 

2. How was this tested:

Forked this repo to my personal GH account and tested there
* The "Bump Package Version" action in my fork: https://github.com/rossedfort/temporal-ui/actions/runs/3715782907
* The resulting PR: https://github.com/rossedfort/temporal-ui/pull/34
* The "Publish NPM Package" action in my fork: https://github.com/rossedfort/temporal-ui/actions/runs/3715803312/jobs/6301396780
* The resulting release: https://github.com/rossedfort/temporal-ui/releases/tag/v4.0.0
3. Any docs updates needed?

### CODE REVIEW

File: .github/workflows/bump-package-version.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Bump version
+        run: npm run package:${{ inputs.bump }}
+      - name: Create Pull Request
+        run: |
+          tag=${{ git describe --abbrev=0 --tags }}
+          branch="releases/${{ tag }}"
+          git checkout -b ${{ branch }}
+          git remote set-url origin https://x-access-token:${{ secrets.COMMANDER_DATA_TOKEN }}@github.com/${{ github.repository }}
+          git push --tags origin ${{ branch }}
+          gh pr create -B main -H ${{ branch }} --title 'bump version in package.json to ${{ tag }}'
```
Comment (rossedfort) [inline .github/workflows/bump-package-version.yml:36]: https://docs.github.com/en/actions/using-workflows/using-github-cli-in-workflows

File: .github/workflows/publish-package.yml [hunk 2]
```diff
@@ -1,45 +1,44 @@
...
       - main
-    inputs:
-      bump:
-        description: 'bump type, major or minor or patch'
-        default: 'patch'
 
 jobs:
-  release:
+  publish:
+    if: github.ref_type == 'branch' && contains(github.ref_name, 'releases/')
```
Comment (rossedfort) [inline .github/workflows/publish-package.yml:8]: With `on: push` I'm not sure if there's a way to omit merges from PRs other than manually checking `ref_type` and `ref_name`. Ideas welcome.

File: .github/workflows/publish-package.yml [hunk 3]
```diff
@@ -1,45 +1,44 @@
...
-      bump:
-        description: 'bump type, major or minor or patch'
-        default: 'patch'
+  pull_request:
+    types: [closed]
 
 jobs:
-  release:
+  publish:
+    if: startsWith(github.head_ref, 'releases/') && github.event.pull_request.merged == true
```
Comment (rossedfort) [inline .github/workflows/publish-package.yml:9]: This is the way GitHub recommends running a workflow when a PR merges. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges

**Comment by rossedfort** [comment]
The "Bump Package Version" action in my fork: https://github.com/rossedfort/temporal-ui/actions/runs/3714727315
The resulting PR: https://github.com/rossedfort/temporal-ui/pull/30
The "Publish NPM Package" action in my fork: https://github.com/rossedfort/temporal-ui/actions/runs/3714755914
The resulting release: https://github.com/rossedfort/temporal-ui/releases/tag/v3.0.8

**Review by stevekinney — APPROVED** [review]

**Comment by rossedfort** [comment]
I just realized this doesn't work with squashing commits on PR merges - going to have to rework it a bit.

**Pull request merged**